### PR TITLE
tree: more efficient expand/collapse buttons

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Closes https://github.com/cylc/cylc-ui/issues/1617
* There is one expand/collapse button per tree item.
* There are many tree items in the tree view.
* We were using two Vuetify components to implement the expand/collapse button, these components are expensive to mount.
* This PR switches to using native components to implement the expand/collapse button.
* This saves ~2/5 of mount+render time.

To test:
* Start the workflow from https://github.com/cylc/cylc-ui/issues/1614 with `hours` set to `20` in paused mode.
* Open the tree view on this workflow and wait for the UI to stabilise.
* Time how long it takes to open a second tree view.

Results:
* Before: ~5s
* After: ~2.5s

Caveats:
* This approach does not support standard Vuetify things like the mouseover "ripple". Personally I see this as more of a pro than a con but it's a change either way.
* This approach does not re-implement the rotate animation on state change. This would be fairly easy to do with CSS, personally I'm happier without, but it's another change.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Review by wxtim — APPROVED** [review]
- [x] Read
- [x] Profiled

File: src/styles/cylc/_tree.scss [hunk 1]
```diff
@@ -55,14 +55,20 @@ $icon-size: 1.2em;
...
   display:inline-block;
   vertical-align:middle;
   width: 100%;
   .node {
     line-height: $line-height;
     display: block;
 
+    .node-expand-collapse-button {
+      width: 1.3em;
+      height: 1.3em;
```
Comment (MetRonnie) [inline src/styles/cylc/_tree.scss:68]: Alignment is slightly off

image

This fixes it:
```suggestion
      width: 1.5em;
      height: 1.5em;
      margin: 0 2px;
```

File: src/components/cylc/tree/TreeItem.vue [hunk 2]
```diff
@@ -18,29 +18,26 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
       :class="nodeClass"
       :style="nodeStyle"
     >
       <!-- the node's left icon; used for expand/collapse -->
-      <v-btn
+      <svg
         v-if="renderExpandCollapseBtn"
         aria-label="Expand/collapse"
-        aria-hidden="false"
         class="node-expand-collapse-button flex-shrink-0"
```
Comment (MetRonnie) [inline src/components/cylc/tree/TreeItem.vue:32]: This automatically adds the rotation transition back in, which I feel is a useful bit of visual feedback
```suggestion
        class="node-expand-collapse-button flex-shrink-0 v-btn"
```

**Review by markgrahamdawson — APPROVED** [review]

File: src/components/cylc/tree/TreeItem.vue (hunk 2 above)
Comment (oliver-sanders) [reply src/components/cylc/tree/TreeItem.vue:32]: Not sure it's a good idea to apply v-btn styling to something which isn't even a `<button />`.

I'm not overly keen on the animation anyway, but would be happy to add this with CSS if desired.

**Review by MetRonnie — COMMENTED** [review]
I have fixed the indentation using relative units at https://github.com/oliver-sanders/cylc-ui/pull/111

(If you combine #1642 with it, everything aligns properly)

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**